packages/opentelemetry-sdk-trace-base/src/Span.ts

```diff
@@ -47,7 +48,7 @@ export class Span implements api.Span, ReadableSpan {
   readonly links: api.Link[] = [];
   readonly events: TimedEvent[] = [];
   readonly startTime: api.HrTime;
-  readonly resource: Resource;
+  readonly resource: IResource;
```

Comment:
Please note that this may be seen as breaking.

Reply:
it is backward compatible but not forward compatible right? I think that's probably ok.

Reply:
before it was possible to access the constructor, now it's no longer possible.